Homework2/ArrayExtension.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Linq;
+
+namespace Homework2
+{
+    public static class ArrayExtension
+    {
+        public static T[] AddAt<T>(this T[] array, T item, int idx)
+        {
+            int newSize = array.Length + 1;
+
+            T[] output = new T[newSize];
+
+            for (int i = 0; i < newSize; i++)
+            {
+                if (i == idx)
+                {
+                    output[i] = item;
+
+                    for (int j = idx; j < array.Length; j++)
```

Comment:
Minor nit: This works but it is maybe not the most intuitive approach to solving this problem.  You could do this all with one loop or with two loops one after the other.  It works so I'm not complaining, just observing.

Reply:
Actually wanted to do some refactoring in the last couple of days but did not find the time to do so :D